### PR TITLE
python3-matplotlib: update to 3.2.1

### PR DESCRIPTION
**Pull request by st3r4g**
Patch refreshed. It works, however on execution it spits out:

> MatplotlibDeprecationWarning: Matplotlib installs where the data is not in the mpl-data subdirectory of the package are deprecated since 3.2 and support for them will be removed two minor releases later.

so I think we should move the data to upstream's preferred location `site-packages/matplotlib/mpl-data`, which should simplify the template and get rid of the patch.

### CODE REVIEW

**Comment by st3r4g** [comment]
> WARNING: The pip package is not available, falling back to EasyInstall for handling setup_requires/test_requires
might be the problem with cross

**Comment by ahesford** [comment]
As we discussed on IRC, I agree that the `python3-matplotlib-data` package should put its files in `${py3_sitelib}/matplotlib/mpl-data` like you suggest, and also update `python-matplotlib-data` to put its files in `${py2_sitelib}/matplotlib/mpl-data`.

Also, remove `replaces="python-matplotlib-data<3.0.0"` from `python3-matplotlib-data` since it won't be a proper replacement (we couldn't guarantee that the files installed by this package were identical or even compatible with the earlier version of matplotlib in `python-matplotlib-data` anyway).

Note that both `python{,3}-matploblib-data` install `/etc/matplotlibrc` as a `conf_file` and patch the `matplotlib` python package to fall back to this file as the last-resort configuration file. I don't think this constitutes a true conflict, since XBPS should just ignore conflicting `conf_files`. However, I'm not sure the current behavior is correct because there is no guarantee that the `matplotlibrc` for the py2 version of `matplotlib` is compatible with the py3 version (and, even if they are compatible now, they could diverge in the future). I think there are three options here:
1. Do nothing, letting both packages "fight" over `/etc/matplotlibrc` and leaving the user to resolve any incompatibilities;
2. Modify current behavior to make `python-matplotlib` look for `/etc/matplotlibrc.py2` and `python3-matplotlib` look for `/etc/matplotlibrc.py3` (and modify their `-data` sub-packages to `vconf` the default configuration into the corresponding locations).
3. Remove the custom behavior altogether, which breaks `/etc/matplotlibrc` but is consistent with upstream behavior and still allows for per-user configurations.

With either option (2) or (3), I think an `INSTALL.msg` should be added to the py2 and py3 packages to note the behavior change.

Personally, I think the third option---relying on standard upstream configuration behavior---is the right choice. Users can always keep a "default" `/etc/matplotlibrc` that can be copied into per-user locations as desired.

**Comment by st3r4g** [comment]
Went for option (3). I don't really have any interest in modifying the py2 package, as it is deprecated anyway and I don't use it. Can we just leave it as it is? I believe there are no conflicts.

This will fix https://github.com/void-linux/void-packages/issues/15751

The sed line about `qhull` is now useless, and I believe the `libqhull-devel` installed by xbps is not being used. Any suggestions for this?

**Comment by ahesford** [comment]
I think I got `python3-matplotlib` to build (and cross-build, at least on `aarch64`) with the system `libqhull`. It probably needs more testing. The changes necessary to make this work are in [PR #1](https://github.com/st3r4g/void-packages/pull/1) on your fork of `void-packages`.

Alternatively, you can just remove the `libqhull-devel` from `makedepends` and remove the entire `pre_build` function, letting `python3-matplotlib` build its own in-tree copy of `qhull`. Your choice!

**Comment by st3r4g** [comment]
As using the system `libqhull` requires a patch, I think it's safer to let it build its own as it's doing now

File: srcpkgs/python3-matplotlib/INSTALL.msg [hunk 1]
```diff
@@ -1,3 +1,12 @@
+NOTICE: /etc/matplotlibrc is no longer supported as configuration file
```
Comment (Chocimier) [inline srcpkgs/python3-matplotlib/INSTALL.msg:1]: 25-line won't even fit into nvidia's 80x25 TTY. Could it be shorter?

Comment (st3r4g) [reply srcpkgs/python3-matplotlib/INSTALL.msg:1]: What about leaving just the first line?
`NOTICE: /etc/matplotlibrc is no longer supported as configuration file`

Comment (st3r4g) [reply srcpkgs/python3-matplotlib/INSTALL.msg:1]: I could put it below, in the part about the backend

Comment (Chocimier) [reply srcpkgs/python3-matplotlib/INSTALL.msg:1]: ok, the rest can be `vdoc`'ed

Comment (st3r4g) [reply srcpkgs/python3-matplotlib/INSTALL.msg:1]: what was the convention for the file to `vdoc`? `README.voidlinux`?

Comment (Chocimier) [reply srcpkgs/python3-matplotlib/INSTALL.msg:1]: yes, `README.voidlinux`

Comment (st3r4g) [reply srcpkgs/python3-matplotlib/INSTALL.msg:1]: done

**Pull request merged**